find.go

```diff
@@ -473,18 +473,33 @@ func Nth[T any, N constraints.Integer](collection []T, nth N) (T, error) {
 	return collection[l+n], nil
 }
 
+// randomIntGenerator is a function that should return a random integer in the range [0, n-1]
```

Comment:
```suggestion
// randomIntGenerator is a function that should return a random integer in the range [0, n)
```

this is the right math notation